segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessorTest.java

```diff
@@ -1506,6 +1521,128 @@ public void testReadTableEntriesDelta() throws Exception {
+        assertEquals(segment, created.getSegment());
+
+        UUID uuid = UUID.randomUUID();
+        WireCommands.AppendSetup setup = (WireCommands.AppendSetup) sendRequest(channel, new WireCommands.SetupAppend(2, uuid, segment, ""));
+
+        assertEquals(segment, setup.getSegment());
+        assertEquals(uuid, setup.getWriterId());
+
+        data.retain();
+
+        //Append 40 bytes of data
+        sendRequest(channel, new Append(segment, uuid, 1, new WireCommands.Event(data), 1L));
+        sendRequest(channel, new Append(segment, uuid, 2, new WireCommands.Event(data), 1L));
+
+        //Total length
+        assertEquals(store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getLength(), 40);
+
+        //Validating the data append data
+        WireCommands.SegmentRead result = (WireCommands.SegmentRead) sendRequest(channel, new WireCommands.ReadSegment(segment, 0, 20, "", 1L));
+        ByteBuf resultAfterOneAppend = result.getData();
+        assertEquals("Hello world", resultAfterOneAppend.toString(Charset.defaultCharset()).trim());
+
+        // Truncate one event
+        final long truncateOffset = 20;
+
+        //Before truncation validation
+        assertEquals(0, store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+        assertEquals(0, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+
+        //After truncation validation
+        sendRequest(channel, new WireCommands.TruncateSegment(requestId, segment, truncateOffset, ""));
+
+        assertEquals(truncateOffset, store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+        assertEquals(0, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+
+        // Truncate at the same offset - verify idempotence.
+        sendRequest(channel, new WireCommands.TruncateSegment(requestId, segment, truncateOffset, ""));
+        assertEquals(truncateOffset, store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+        assertEquals(0, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
+
+        //Deleting the main segment and validating that index segment has also deleted
+        sendRequest(channel, new WireCommands.DeleteSegment(1, segment, ""));
+        assertThrows(StreamSegmentNotExistsException.class, () -> store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join());
+        assertThrows(StreamSegmentNotExistsException.class, () -> store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join());
+    }
+
+    @Test
+    public void testTruncateIndexSegmentWithNegativeOffset() throws Exception {
```

Comment:
Is test scenario of doing multiple appends(more then 3 ) and then truncating at least 2 segment covered, checking the offsets of main and index segment covered?